src/azul/service/avro_pfb.py

```diff
@@ -0,0 +1,451 @@
+                                                            },
+                                                            {
+                                                                "name": "values",
+                                                                "type": {
+                                                                    "type": "map",
+                                                                    "values": "string",
+                                                                },
+                                                            },
+                                                        ],
+                                                    },
+                                                },
+                                            },
+                                        ],
+                                    },
+                                },
+                            },
+                            {
+                                "name": "misc",
+                                "type": {"type": "map", "values": "string"},
+                            },
+                        ],
+                    },
+                    *azul_avro_schema
+                ]
+            },
+            {
+                "name": "relations",
+                "type": {
+                    "type": "array",
+                    "items": {
+                        "type": "record",
+                        "name": "Relation",
+                        "fields": [
+                            {"name": "dst_id", "type": "string"},
+                            {"name": "dst_name", "type": "string"},
+                        ],
+                    },
+                },
+                "default": [],
+            },
+        ],
+    }
+
+
+def tables_schema(field_types: FieldTypes) -> Iterable[JSON]:
+    """
+    Produce a sequence Avro schemas for each inner entity given a FieldTypes
+    mapping.
+
+    :param field_types: Dictionary structure mapping the ES doc structure to
+        types.
+    :return: A generator of schemas
+    """
+    for entity_type, field_types in field_types.items():
+        # We skip primitive top-level fields like total_estimated_cells
+        if isinstance(field_types, dict):
+            yield {
+                "name": entity_type,
+                "type": "record",
+                "fields": list(_tables_schema_recursive(field_types, files_entity=entity_type == 'files'))
+            }
+
+
+nullable_to_pfb_types = {
+    null_bool: ['string', 'boolean'],
+    null_float: ['string', 'double'],  # Not present in current field_types
+    null_int: ['string', 'long'],
+    null_str: ['string'],
+    null_int_sum_sort: ['string', 'long']
+}
+
+
+def _tables_schema_recursive(field_types: FieldTypes, files_entity: bool = False) -> Iterable[JSON]:
+    for entity_type, field_types in field_types.items():
```

Comment:
```suggestion
    for entity_type, field_type in field_types.items():
```

The scope during which this is a list is much smaller than the scope during which it isn't so it would make more sense to use a variable name that fits the larger scope. Or use two variables.

Reply:
`field_types` may also be a dict, it which case I think the plural name is more appropriate. None the less, most of the cases are for "leafs" so I'll move to `field_type`.